### DISCUSSION TITLE
Welcome to DIY-Multiprotocol-TX-Module Discussions!

### DISCUSSION THREAD

**Discussion by pascallanger** [pascallanger/DIY-Multiprotocol-TX-Module#479] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Gaylian** [comment]
I'm new to this discussion. I have a Jumper T16 Pro with Multiprotocol module. I usually use FRSKY D16 and Jumper R1 Rx's. I have a FRSKY TFR6 Rx I was hoping to use but it seems as though there is no FASST protocol for this RX. My question is does the module support FASST protocol.

**Reply by ANS-HAFEEZ** [reply]
Hi, I also have jumper R1 D16 receiver but it won't bind with my multi-protocol module. After some search I found that it has D16 v2.1 and can't bind with mutiprotocol module.
Can you please guide me in this regards?
Thanks:)

**Comment by pascallanger** [comment]
Futaba FASST is not supported and won't be since the multi module does not have the hardware to add it.

**Reply by wollenagold** [reply]
Hy, i'm New in this Diskussion and my question is the Support of the fhss-Protokoll of the carson wheel pro tx. I didnt find a protokoll and posibility to bind the iRange x-Modul i use in my frsky x-lite s with the receiver carson Reflex wheel pro 4ch-rx. I searched the Web but i did not find anything that helps me.
Thanks for a helpfull answer if that is possible or not.

**Reply by pascallanger** [reply]
The Carson protocol is not supported.

**Comment by Duncandan1** [comment]
Hi I'm Dan, been flying RC for a number of years mostly glider and powered gliders.I've been using a Taranis X9D for 6-7 yrs but decided to buy the Radiomaster T16S. Open TX makes it great because I know a bit about it. I recently bought a toy style drone and the multiprotocol module is new to me. I can get my FrSky receivers and a Tactic. My question is how do I figure the protocol for a drone that isn't on the list.
 https://hobbyking.com/en_us/2-million-pixel-camera.html?___store=en_us  Thanks DAN

**Reply by pascallanger** [reply]
Hi, No idea what protocol it can use or if it's even supported...
2 options:
 - Write on rcgroups to see if someone as an idea: https://www.rcgroups.com/forums/showthread.php?2165676-DIY-Multiprotocol-TX-Module
 - Try every protocol and subprotocols

**Reply by Duncandan1** [reply]
Thanks pascallanger, shall do

**Comment by vojtas81** [comment]
Hi, I want to use my TX16S to fly with Hubsan Spy Hawk 301s and everything seems to work fine. The only issue I have is, that my transmiter doesn`t receiwe any telemetry data. My question is, am I doing something wrong, or there are simplny no telemtery data sended in to my transmiter? Thank you

**Reply by pascallanger** [reply]
Hubsan telemetry should work for simple things like the battery and ...
Have you launched a sensor discovery?

**Reply by vojtas81** [reply]
Thank you for fast reply. Yes, I have launched sensor discovery, but without success. Do you mean "Detect new sensors" on "Telemetry" page?

**Reply by pascallanger** [reply]
Yes, that's where they should appear. The discovery should be launched when the quad is powered and bound to the radio.

**Reply by vojtas81** [reply]
Ok, that's what I do. I have no idea what might be wrong then. Also transmitter and multi module is updated.

**Reply by pascallanger** [reply]
Do you see the multi module version and status where you select the protocol?

**Reply by vojtas81** [reply]
Sorry for language, it is Czech version.
![DSC_0268](https://user-images.githubusercontent.com/78214946/106308694-5b6aae80-6261-11eb-80ec-3bf830cbb62c.JPG)

**Comment by FPVxfun** [comment]
Dear Pascal, is it possible to add FAILSAFE to DSM (Spektrum) protocol?

Thanks in advance

**Reply by pascallanger** [reply]
The DSM protocol does not allow failsafe to be transmitted.
If your RX is compatible with DSM forwarding then use [this LUA script](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/tree/master/Lua_scripts#dsm-forward-programming) otherwise failsafe is done on the RX itself.

**Comment by Wesie** [comment]
Hi,
first at all: Thx a lot for your great work. 
Uncomfortable I couldn't found any information, which protocol is incl. in the precompiled FW on the DL Side. In my special case the question is more relevant for the 4:1 AVM Modul, because the memory is limited and also in history only a subset of protocols found place. 
It will be a nice idea to have this information on the Download page, because of the included or excluded protocols the  decision must make for DL and flash or complied by his own

**Reply by pascallanger** [reply]
It's in the file name. The A7105 build has all the protocols running with this RF component. The cyrf build has all the protocols running with this RF component. Same for the cc2500.
These are mostly test builds.
Compiling your own version for an avr module is a must.

**Comment by kvdelsen** [comment]
Hey there! Just a quick question: is frsky ACCESS coming to mpm? And is there any firmware available on this github page for my RM TX16S? I could only find firmware on the MultiProtocolModule website which was labeled for it...

**Reply by pascallanger** [reply]
https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/issues/487
https://www.multi-module.org/using-the-module/firmware-updates

**Reply by kvdelsen** [reply]
Oh allright, so that site does have the correct version, was a bit confused wether firmware would be here or there, thanks! How about the access support?

**Reply by pascallanger** [reply]
https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/issues/487

**Comment by adso85swe** [comment]
Hi Pascal. Great work with the multiprotocol support! Amazing stuff.

I am wondering about the reson for the lack of failsafe on the WFLY protocol. Are there technical limitations for such support or is it simply due to lack of demand from the users?

**Reply by pascallanger** [reply]
I'm not aware that you can set failsafe from the transmitter with WFLY. It's set on the RX if I'm not wrong, isn't it?
Failsafe seeing from the radio is supported with WFLY2.

**Reply by adso85swe** [reply]
I own a original WFLY WFT07 transmitter. It does indeed allow you to put individual channels on either hold, or a specific output value in percentage. This is done i a dedicated failsafe section in the menu. 

The reason I'm asking is because I have been testing a new TS16X transmitter with my old WFLY receivers. When I simulate a lost signal by turning off my transmitter, the servos go to neutral points, which is fine really. But being able to set a custom failsafe would be sweet if possible.

Another issue I have had which is semi related; after having turned off the transmitter and turning it back on again, it does not always bind with the receiver again automatically. I have to go back to the binding menu and rebind the model. Not sure if this is a SW issue. It has only happened during these signal loss simulations by turning the radio completely off. Maybe that's not really representative for a real signal loss? I can't decide if I dare to use my old WFLY receivers with this issue 😄

**Reply by pascallanger** [reply]
> I own a original WFLY WFT07 transmitter. It does indeed allow you to put individual channels on either hold, or a specific output value in percentage. This is done i a dedicated failsafe section in the menu.

By soldering a logic analyzer (cheap on ebay) on the SPI bus in your WF07, we could add the missing failsafe by doing some dumps. Let me know if you are willing to do it.

> Another issue I have had which is semi related; after having turned off the transmitter and turning it back on again, it does not always bind with the receiver again automatically. I have to go back to the binding menu and rebind the model.

Having to do a bind again is highly surprising. The RX should remember the TX and the TX is always using the same ID so...
Are you sure you haven't enabled bind on channel for example? or you are not changing the RX number?

**Reply by pascallanger** [reply]
Looking at the code, turning off the radio would be totally different from a signal loss from the RX perspective. On the original module (I worked on a WFLY module  with limited features not a radio) and what I've reproduced, each time you reboot the TX it randomely selects the RF channels which are going to be used and send them to the RX. If you turn off the TX but not the RX, I guess the RX will just continue to use the old RF channels and won't reconnect since they are now different...

**Reply by adso85swe** [reply]
Hmmm, I'm continuing my experimenting with the WFLY protocol and found some new interesting conflicts.
I have binded the Tx to two different Rx, and they are working fine individually as long as I don't do that "signal loss simulation" as I described above, which is not really representative anyway according to your code check. However, I still get problems if I keep the Tx on at all times and switch airplane model to the other Rx, then I need to rebind this Rx again although it worked previously. Could this also be related to the that the receivers may have been binded with different random RF channels and the Tx is not able to swap between these when changing model in the Tx software? 

Regarding that logic analyzer, sure, if it will help the community I could try to do that. Could you post a link to the device and I will look into purchasing it, as a start. 

Edit: Btw, when switching between those two receivers, I do disconnect the first one from power before switching the airplane model in the Tx, and lastly I power up the second Rx.

**Reply by adso85swe** [reply]
Oh, and I also wanted to mention just for your information in case that helps for future code optimisations, I have tried those signal loss simulation Tx power downs with the original WFLY transmitter many times in the past, and that combination never showed these issues with rebinding needs. So this is something unique to the multi protocol module/code for sure.

**Reply by pascallanger** [reply]
Ok let's review the protocol when you'll be able to do some dumps. You are the first one to report issues but may be no one has tested it like you before... For the logic analyzer, search on ebay for "saleae 8ch analyzer". When you are ready open an issue and we will work on it.

**Reply by adso85swe** [reply]
Sounds good.
I also own Radiomaster receivers using the FrSky D8 protocol and I tried these today in the exact same way as with the WFLY receivers. The Radiomaster receivers worked flawlessly, always rebinding automatically no matter what I did.

To my surprise the D8 protocol does not have failsafe support either, but that's not that big of a deal I geuess. The most important part for me is that it works solidly, in contrast to the WFLY.

I might sell my complete WFLY kit to get some money back from my Radiomaster purchases, and in that case I might not be able to do the protocol data extraction. But on the other hand it would have been nice to help the community out. I'll think about it. The logic analyser is dirt cheap indeed.

**Reply by pascallanger** [reply]
Yes D8 does not support failsafe from the radio, D16 does.
I'll see if I can reproduce your issue as I think I still have a WFLY RX. You just power off/on the radio and the RX will not reconnect every time, correct? You even say that sometimes you have to launch a bind on the TX and RX otherwise it won't reconnect at all?
You can set the failsafe when you want or it is happening at bind time? If you set failsafe on a RX on your WF07 and then bind it on Multi, the failsafe information is gone, correct?

**Reply by adso85swe** [reply]
Yes, the failsafe that I set on the WFLY transmitter on a certain receiver is gone when I bind that receiver to the multi module. I can not set any failsafe mode directly on the receiver, so when I turn off the multi module it seems that the servos go to neutral positions, except for the throttle which stays turned off att -100%. 

Correct, when I turn the multi transmitter off/on, then approximately 25% of the times the receiver does not rebind by itself. When this happens the only solution is to rebind manually, and even this can be tricky. Sometimes during the rebinding procedure it connects for about 3 seconds and then looses connection again. But after a few attempts, sometimes powering everything off/on, rebinding is eventually possible.

**Reply by pascallanger** [reply]
I'm wondering then if this failsafe info is not sent during bind if it gets overwritten by this operation. This is the only place where there are a number of unknown bytes. I will see if I can change the bytes and get a different failsafe pattern.
I should be able to reproduce the issue "easily" if it happens 25% of the time.

**Reply by pascallanger** [reply]
@adso85swe I've implemented failsafe to WFLY. Please test using this version v1.3.2.49: https://downloads.multi-module.org/latest-test/. As you may know failsafe in OpenTX is sent every 10 seconds or so, be patient after setting up the failsafe values (values and not Hold or None) before testing.
I haven't been able to reproduce your reconnection issues by using my module and powering off/on more than a dozen of times. I'm wondering then if this is based on your ID since this is the only thing different. I'll try to do more testing...

**Reply by pascallanger** [reply]
Can you confirm how you reproduce the issue?
On my side I have a WFR07S RX powered on with a servo on CH1. On OpenTX, I've setup a servo sweep on CH1. I'm powering on/off the radio again and again with no issues. The servo goes from the failsafe position a few seconds after the power off. As soon as the radio/module is powered on the servo sweep restarts.
I've done so many trials now to reproduce that I don't know where to look at...

**Reply by adso85swe** [reply]
Oh, that was a quick update. Cool. I will test it out later this week, probably during the weekend. 

Your trials to reproduce my issues sounds correct, just like I did it. I'm starting to wonder if the issues are related to the fact that I binded to two different receivers, a WFR06S and a WFR04S. Perhaps that created a conflict? I'm not completely sure exactly when the problems started to appear, before or after bindind that second receiver. I do remember that I did not have any issues from the very beginning, but I'm not sure exactly which event triggered it eventually. I will do some more trials with my current SW and with your new SW. If I spot the issue again I will try to film it.

**Reply by pascallanger** [reply]
I haven't changed anything on the firmware which would address your connectivity issue ref "I will do some more trials with my current SW and with your new SW."
I've checked the code but unless I can reproduce the issue it's going to be hard to figure out what's happening. Like I said, the only thing I could think of is the ID but since you have a TX16S I don't know how to read your ID without connecting a serial adapter to your internal module. If you have the hardware (3.3V FTDI) and are ready to do that let me know. I'm thinking to add this feature to OpenTX at some point but it's not for tomorrow.

**Reply by adso85swe** [reply]
Hi. I have recorded a video to demonstrate the rebinding issues I'm having with WFLY. How can I share it with you? Do you use OneDrive?

**Reply by adso85swe** [reply]
If you could give me your email address I can send you a link to the video. Or let me know if you have other suggestions.

**Reply by pascallanger** [reply]
One drive is fine. But I'm not sure seeing the video will help troubleshooting...

**Reply by adso85swe** [reply]
Let me know if you have managed to see it. I will probably remove the link after that. I guess you can download the video locally for your own reference.

**Reply by pascallanger** [reply]
I can't see it. I thought it was my mobile phone yesterday evening but same on the PC this morning.

**Reply by pascallanger** [reply]
Ok I can see the video now.
You do not rebind the RX as you previously stated... 
When you launch a bind as you do, you just relaunch the protocol which is the same as turning off/on the radio.
To emulate the radio on/off you can either launch a bind or change the RX number back and forth 0,1,0,1,0...
My guess is that your RX does not like some of the frequencies being used. Since they are selected randomly at each start up sometimes it works and not other with no direct correlation.
The RX I have do not do the same. I still don't know if this is linked to the ID or not.
The only way to troubleshoot further is to connect a FTDI to your module to see which frequencies are causing issues and if I use your ID I can reproduce the behavior.

**Reply by adso85swe** [reply]
Interesting. I am not an expert in how these Tx/Rx protocols work by any means. If the Rx doesn't like some of the frequencies chosen by the Tx, does it mean that the emulation of the WFLY protocol in the Tx is slightly incorrect? I.e., does the Tx choose frequencies that are not supported in the "real" WFLY protocol?

No matter what the technical problem is, with this behaviour, do you still think that there is no risk with triggering such a frequency mismatch during flight?

Btw, just to be clear, I did not yet try your new software that supports custom failsafe setting. This was still the latest official release.

**Comment by ghost** [comment]
Thank you for providing a great mechanism and I am having a good radio life.

I really like the quick response of the REDPINE protocol and the feel of the futaba 18sz radio gimbal. So, I would like to connect jp4in1 to futaba radio and use REDPINE protocol.
I was able to connect jp4in1 to the futaba radio with PPM and use the REDPINE protocol, but I could not get the quick response of the REDPINE protocol. So, I would like to connect jp4in1 to futaba radio in serial mode. 

I understood the connection method for serial connection by referring to the CRSF example, but futaba radio cannot send protocol specification or bind command to jp4in1 in serial mode. Is there a way to connect and use jp4in1 in serial mode with futaba radio?

I thought that it could be solved by editing config.h in the case of serial mode on the jp4in1 side and specifying to use the REDPINE protocol, but I did not know how to realize it. .. Thank you.

**Reply by pascallanger** [reply]
No idea what a Futaba radio can output... Do they have specs somewhere?

**Reply by kilrah** [reply]
PPM is the only thing a T18SZ can output and it'll obviously be slow, so time to get used to another radio...

**Reply by ghost** [reply]
Thank you for reply!
I'm not familiar with it, but futaba has a serial protocol called s-bus. 
https://os.mbed.com/users/Digixx/notebook/futaba-s-bus-controlled-by-mbed/

I wanted to use a radio that I was familiar with. 
But I may have been too greedy.
Thank you.

**Reply by pascallanger** [reply]
The SBUS they are talking about here is the output of the RX not the TX...

**Reply by pascallanger** [reply]
And adding the module behind a few will increase the latency even more so no point in doing this.

**Reply by ghost** [reply]
I am very happy to have your support.
I was able to understand well.
Thank you.

**Comment by wollenagold** [comment]
Thanks for the  Information. Change the RX



Holen Sie sich Outlook für Android

**Comment by asaerong** [comment]
Hello, I am Albert and new here in the discussion. I just wanted to say thanks to Pascal for the great work on multiprotocol! and hoping to get advice on my case. 

I was using a Jumper T-Lite and the internal multi module works great, and I can easily clone my old XJT JR module and use Frsky D8 cloned and FrskyX D16 cloned with my old quads (DII4R and XSR). But I recently moved to use TBS Tango 2 and bought TBS MPM with the thought that I will have the same experience as with my jumper which apparently it wasn't. Both are at the latest firmware (mpm 1.3.2.77, tango 1.3.29). cloning the TX works fine but FrskyD cloned sub type is missing in TBS MPM version, also FrskyX  (skipped). The only available clone subtype is in FrskyX2, which is can't be used for my purpose. 

Pascal stated in rcgroups about 7 months ago that it was freedomtx (tango)'s problem because of they are using old multicode, which seems hasn't been fixed until now, in the meantime, seems that their github was already abandoned because it's last release posted there was 1.1.3 while actually the latest recent version supposed to be 1.2.9. so not sure raising the issue there will be read by them.. is there way to make it work or a workaround from mpm side? or it is just a given that until freedomtx fixed their thing, Frsky radio cloning feature does not fully work in TBS Tango 2+mpm....  

Worst case, i'll re-bind everything (9 of them :) LoL) with tbs mpm and make it as a master, and have the jumper as mybackup to clone the mpm instead of my old XJT. but it would be great if I don't have to go that route :)

**Reply by pascallanger** [reply]
You posted the same question on RCG and I've answered there.

**Reply by asaerong** [reply]
Yep, got it. Thanks Pascal!

**Comment by Winslow87** [comment]
Hi Pascal,
I use a Horus X12S with external iRangeX+ module ( firmware v1.3.2.77).
All worked fine 'til OpenTX v2.3.11 but since updating to v2.3.12, I get a message "Invalid protocol" when selecting "MULTI". Downgrading to 2.3.11 fix the problem.
Any idea ?
Thanks for your nice job.
Pascal

**Reply by pascallanger** [reply]
Upgrade your module to v1.3.2.86: https://downloads.multi-module.org/latest-test/

**Reply by Winslow87** [reply]
Thanks for your help.

I've upgraded multi module to v1.3.2.86 but that don't fix the issue for OpenTx versions higher than 2.3.11 (message "invalid protocol" instead "module fw version" in "Module status" box and the protocol is stuck on "Assan".
Note : if the module is not present, it is possible to scroll throught all protocols an "Module status" is "No telemetry...".
I launched "Multi module" tool in "system/tools" menu : I got the message "Module not configured" or something like that (my Tx language is french).
I 've done the same test after downgrading OpenTx to v2.3.11 : the message is "External Multi detected - Protocol "name of the selected protocol" - Channels..."

Regards
Pascal
